File: front/src/models/Projeto.ts
import Cliente from './Cliente';
import Atividade from './Atividade';
import Status from './Status';

export default class Projeto {
  id: string;
  nome: string;
  status: Status;
  cliente: Cliente;
  atividades: Atividade[];

  constructor(id: string, nome: string, status: Status, cliente: Cliente, atividades: Atividade[] = []) {
    this.id = id;
    this.nome = nome;
    this.status = status;
    this.cliente = cliente;
    this.atividades = atividades;
  }
}

File: back/src/main/java/com/agcapital/domain/Projeto.java
package com.agcapital.domain;

import java.util.List;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Projeto {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Column(nullable = false)
    private String nome;

    
    @ManyToOne
    @JoinColumn(name = "status_id" )
    private Status status;

    @ManyToOne
    @JoinColumn(name = "cliente_id", nullable = false)
    @JsonIgnoreProperties("projetos")
    private Cliente cliente;

    @OneToMany(mappedBy = "projeto", cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<Atividade> atividades;
}
